### PR TITLE
Change entitlement check to return future.success only if authorized.

### PR DESCRIPTION
**Pull request by rabbah**
pg3/157.

### CODE REVIEW

File: core/controller/src/main/scala/whisk/core/controller/AuthorizedRouteDispatcher.scala [hunk 1]
```diff
@@ -67,18 +65,16 @@ trait BasicAuthorizedRouteProvider extends Directives with Logging {
...
     }
 
-    protected def handleEntitlementFailure(failure: Try[Boolean])(
+    protected def handleEntitlementFailure(failure: Throwable)(
         implicit transid: TransactionId): RequestContext => Unit = {
         failure match {
-            case Success(false)            => terminate(Forbidden)
-            case Failure(r: RejectRequest) => terminate(r.code, r.message)
-            case Failure(t)                => terminate(InternalServerError, t.getMessage)
-            case _ /*Success(true)*/       => terminate(InternalServerError, "Should not reach here.")
```
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/AuthorizedRouteDispatcher.scala]: removes this silly quirk.

File: core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala [hunk 2]
```diff
@@ -205,6 +205,10 @@ protected[core] abstract class EntitlementProvider(config: WhiskConfig, loadBala
                 info(this, s"not authorized: $r")
             case Failure(t) =>
                 error(this, s"failed while checking entitlement: ${t.getMessage}")
+        } flatMap {
+            if (_) {
```
Comment (markusthoemmes) [inline core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala]: minor: want to provide a variable name like `isAuthenticated`? This reads a bit odd.

Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala]: https://github.com/openwhisk/openwhisk/pull/1722/commits/7f5b44be5d2c254127fe2e720fd28e8cc537ca50

**Comment by rabbah** [comment]
rebased to master.

**Comment by psuter** [comment]
Merged as de9cc68db54499eee8dc802ecf34a67c2ce56f33

**Pull request closed without merging**